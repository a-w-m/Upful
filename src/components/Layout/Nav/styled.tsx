import styled from "styled-components"
import device from "../mediaQuery"
import { Link } from "gatsby"

export const Menu = styled.nav<{ open: boolean }>`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: #f4f4f4;
  height: 100vh;
  width: 100%;
  text-align: left;
  position: fixed;
  top: 0;
  left: 0;
  transition: transform 0.3s ease-in-out;
  transform: ${({ open }) => (open ? "translateX(0)" : "translateX(-100%)")};
  //hiding slide-out-navigation to prevent focus from keyboards or screen readers
  visibility: ${({ open }) => (open ? "visible" : "hidden")};
  z-index: 9;

  @media ${device.laptop} {
    flex-direction: row;
    align-items: center;
    position: static;
    height: 2rem;
    transform: none;
    visibility: visible;
    background: none;
    gap: 2rem;
    width: 100%;
  }

  /*Remove logo from hamburger menu*/
  & > a:nth-child(1) {
    display: none;
    @media ${device.laptop} {
      display: block;
    }
  }
`

export const NavLink = styled(Link)`
  font-size: 2rem;
  text-transform: uppercase;
  padding: 2rem 0;
  padding-left: 2rem;
  font-weight: bold;
  letter-spacing: 0.5rem;
  color: #000;
  text-decoration: none;
  transition: color 0.3s linear;

  @media ${device.laptop} {
    font-size: 1.7rem;
    font-weight: 400;
  }
  &:hover {
    color: var(--header-nav-link-hover);
  }
`
