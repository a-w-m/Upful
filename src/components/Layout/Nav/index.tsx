import React from "react"
import { Menu, NavLink } from "./styled"
import Logo from "../Logo"
import { graphql, useStaticQuery } from "gatsby"
import { C } from "../../../interfaces"

interface NavProps {
  open: boolean
}

const Nav: React.FC<NavProps> = props => {
  const { open } = props

  const data: C.MenuLinks = useStaticQuery(categoryQuery)

  return (
    <Menu open={open} id = {"menu"} aria-label = "Main">
      <Logo />
      {data.site.siteMetadata.menuLinks.categories.map(category => {
        return (
          <NavLink to={category.slug} key={category.name}>
            {category.name.toUpperCase()}
          </NavLink>
        )
      })}
    </Menu>
  )
}

export const categoryQuery = graphql`
  {
    site {
      siteMetadata {
        menuLinks {
          categories {
            name
            slug
          }
        }
      }
    }
  }
`

export default Nav
