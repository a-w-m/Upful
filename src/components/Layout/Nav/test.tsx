import React from "react"
import { render, screen } from "@testing-library/react"
import Nav from "."

describe("Nav component", () => {
  it("should match snapshot", () => {
    render(<Nav open={true} />)
    const container = screen.getByRole('navigation')
    expect(container).toMatchSnapshot()
  })

  it("should not be visible when not open", () => {
    render(<Nav open={false} />)
    const navigation = screen.queryByRole('navigation', {hidden: true})
    expect(navigation).not.toBeVisible()
  })

  it("should be visible when open", () => {
    render(<Nav open={true} />)
    const navigation = screen.getByRole("navigation")
    expect(navigation).toBeVisible()
  })

  it("should contain all navigational links", ()=>{
    render(<Nav open={true} />)
    const links =  screen.getAllByRole('link')
    expect(links).toHaveLength(2)
  })

  it("should display Logo", ()=>{
    render(<Nav open={false} />)
    const logo = screen.getByAltText(/logo/)
    expect(logo).toBeInTheDocument()

  })
})
