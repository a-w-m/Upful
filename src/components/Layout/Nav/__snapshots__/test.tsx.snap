// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component should match snapshot 1`] = `
<nav
  aria-label="Main"
  class="sc-bczRLJ fjQfyL"
  id="menu"
  open=""
>
  <a
    class="sc-dkzDqf dkpeqK"
    href="/"
  >
    <div
      class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
      data-gatsby-image-wrapper=""
      style="position: relative; overflow: hidden; display: inline-block; vertical-align: top; border-radius: 100%;"
    >
      <div
        style="max-width: 250px; display: block;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;charset=utf-8,%3Csvg height='250' width='250' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
          style="max-width: 100%; display: block; position: static;"
        />
      </div>
      <div
        aria-hidden="true"
        data-placeholder-image=""
        style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
      />
      <picture>
        <source
          data-srcset="/static/55955cd5db9ac5b9496eb4fb43c87585/74c72/upful-gold-frame-logo.webp 63w,/static/55955cd5db9ac5b9496eb4fb43c87585/d66e1/upful-gold-frame-logo.webp 125w,/static/55955cd5db9ac5b9496eb4fb43c87585/e7160/upful-gold-frame-logo.webp 250w,/static/55955cd5db9ac5b9496eb4fb43c87585/5f169/upful-gold-frame-logo.webp 500w"
          sizes="(min-width: 250px) 250px, 100vw"
          type="image/webp"
        />
        <img
          alt="logo"
          data-gatsby-image-ssr=""
          data-main-image=""
          data-src="/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png"
          data-srcset="/static/55955cd5db9ac5b9496eb4fb43c87585/6a639/upful-gold-frame-logo.png 63w,/static/55955cd5db9ac5b9496eb4fb43c87585/2fd20/upful-gold-frame-logo.png 125w,/static/55955cd5db9ac5b9496eb4fb43c87585/de391/upful-gold-frame-logo.png 250w,/static/55955cd5db9ac5b9496eb4fb43c87585/82c11/upful-gold-frame-logo.png 500w"
          decoding="async"
          loading="lazy"
          sizes="(min-width: 250px) 250px, 100vw"
          style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
        />
      </picture>
      <noscript />
      <script
        type="module"
      >
        const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
      </script>
    </div>
  </a>
  <a
    class="sc-gsnTZi flUqwp"
    href="/women/"
  >
    WOMEN
  </a>
  <a
    class="sc-gsnTZi flUqwp"
    href="/kids/"
  >
    KIDS
  </a>
</nav>
`;
